src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/InitializeElementFacesGridCoordinates.hpp

```diff
@@ -0,0 +1,100 @@
+ * the worldtube each time step. This data is sent to the corresponding element
+ * where it is used to apply pointwise boundary conditions. The
+ * `ElementFacesGridCoordinates` holds a map of all the element ids abutting the
+ * worldtube with the corresonding grid coordinates.
+ *
+ * \warning This currently assumes that initial domain remains the same and
+ * there is no AMR. To support this, the worldtube could send the
+ * coefficients of its internal solution to each element which can evaluate the
+ * it on their current grid in the boundary conditions.
+ *
+ *  DataBox changes:
+ * - Adds: nothing
+ * - Removes: nothing
+ * - Modifies: Tags::ElementFacesCoordinatesMap<Dim>
+ */
+template <size_t Dim>
+struct InitializeElementFacesGridCoordinates {
+  using return_tags = tmpl::list<Tags::ElementFacesGridCoordinates<Dim>>;
+  using argument_tags =
+      tmpl::list<::domain::Tags::Domain<Dim>,
+                 ::domain::Tags::InitialExtents<Dim>,
+                 ::domain::Tags::InitialRefinementLevels<Dim>,
+                 evolution::dg::Tags::Quadrature, Tags::ExcisionSphere<Dim>>;
+  static void apply(
+      const gsl::not_null<std::unordered_map<
+          ElementId<Dim>, tnsr::I<DataVector, Dim, Frame::Grid>>*>
+          element_faces_grid_coords,
+      const Domain<Dim>& domain,
+      const std::vector<std::array<size_t, Dim>>& initial_extents,
+      const std::vector<std::array<size_t, Dim>>& initial_refinement,
+      const Spectral::Quadrature& quadrature,
+      const ::ExcisionSphere<Dim>& excision_sphere) {
+    const auto& blocks = domain.blocks();
+    const auto& worldtube_grid_coords = excision_sphere.center();
+    const auto& neighboring_blocks = excision_sphere.abutting_directions();
+    for (const auto& [block_id, _] : neighboring_blocks) {
+      const auto element_ids =
+          initial_element_ids(block_id, initial_refinement.at(block_id));
+      for (const auto element_id : element_ids) {
+        const auto direction = excision_sphere.abutting_direction(element_id);
+        if (direction.has_value()) {
+          const auto mesh = ::domain::Initialization::create_initial_mesh(
+              initial_extents, element_id, quadrature);
+          const auto face_mesh = mesh.slice_away(direction.value().dimension());
+          const auto& current_block = blocks.at(block_id);
+          ElementMap<Dim, Frame::Grid> element_map{
+              element_id,
+              current_block.is_time_dependent()
+                  ? current_block.moving_mesh_logical_to_grid_map().get_clone()
+                  : current_block.stationary_map().get_to_grid_frame()};
+          const auto face_logical_coords =
+              interface_logical_coordinates(face_mesh, direction.value());
+          auto faces_grid_coords = element_map(face_logical_coords);
+          for (size_t i = 0; i < 3; ++i) {
+            faces_grid_coords.get(i) -= worldtube_grid_coords.get(i);
```

Comment:
Please add to the docs (of this function _and_ the tag) that this actually stores the grid coords _centered on the worldtube_. Seems important :D